src/index.js

```diff
@@ -289,11 +291,12 @@ export default function undoable (reducer, rawConfig = {}) {
           }
         }
 
-        if (history.present !== res) {
+        // Don't add entry to history if it was just created
+        if ((justCreatedHistory && history.present === res) || action.type === '@@redux-undo/INIT') {
```

Comment:
Pff, I've spend some more hours on this but cannot get it right..

I think we do have some big problems with HMR. 
I've added some test that show that replacing reducers is not working how users might expect it. 

See the `distinct_state_fixes_more_fixes` branch: https://github.com/omnidan/redux-undo/commit/29a3e7ebf5335cf81f0005b12598871df88c30b3

Maybe you could check the comments out and let me know what you think. I've added some comments in the code...